.github/workflows/build.yml

```diff
@@ -484,3 +484,77 @@ jobs:
         run: cargo test --locked -- --include-ignored
         env:
           RUST_BACKTRACE: 1
+
```

Comment:
However, I'm afraid I don't like the duplication here. Note we also already deny warnings for all features and no features, which should catch 99% of the issues -- see the clippy job.

Reply:
Maybe we can deny warnings on the powerset job to close some of the remaining gaps?

Reply:
> we also already deny warnings for all features and no features, which should catch 99% of the issues -- see the clippy job

Got it ... guessing the clippy job does not cover the release configuration?

The one issue I found with this was with release + logging combination.

> deny warnings on the powerset job

It looiks to me like the powerset job already does deny warnings, or am I mistaken here?

Incidentally I would like to switch to better spelled-out `--deny warnings` for improved clarity & consistency.

I did try running `cargo hack check` with the feature powerset (only on `--package rustls`) command from daily-tests.yml, seems to fail with this:

```sh
cargo check --no-default-features --features brotli,ring --package rustls
```

I will investigate and hopefully find a way to resolve this soon.

As an aside, I tried running `cargo hack test --package rustls` with the powerset, failed with `custom-package`, everything else seems to be OK. The test issue with the `custom-package` seems like a minor cfg issue in the tests, applied a quick workaround in my workarea for now.